File: python/arrays/961-N-repeated-element-in-size-2N-array.py
# Problem: Leetcode 961 - N repeated element in size 2N array
# Difficulty: Easy
# Link: https://leetcode.com/problems/N-repeated-element-in-size-2N-array/description/
# Time Complexity: O(n)
# Space Complexity: O(n) as we use a set
# Approach: As only one element repeats we only have to return the first duplicate found which is quite straightforward.
# so we keep a set and if an element found in the set we return it immediately.


from typing import List
class Solution:
    def repeatedNTimes(self, nums: List[int]) -> int:
        seen = set()
        for num in nums:
            if num in seen:
                return num
            seen.add(num)

File: Java/arrays/961-N-repeated-element-in-size-2N-array.java
/* 
# Problem: Leetcode 961 - N repeated element in size 2N arrat
# Difficulty: Easy
# Link: https://leetcode.com/problems/n-repeated-element-in-size-2N-array/description/
# Time Complexity: O(n)
# Space Complexity: O(n) as we use a set
# Approach: We just check for the first repeated value and we immediately return. we can have a final return 0 or throw a runtTime exception if 
# if the code flow reaches there as it was not supposed to.
*/
class Solution {
    public int repeatedNTimes(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int num:nums){
            if (s.contains(num)){
                return num;
            }
            s.add(num);
        }
     return 0;
    }
}